field/src/extension/mod.rs

```diff
@@ -8,5 +10,48 @@ pub mod quadratic;
 /// such that the extension is `F[X]/(X^d-W)`.
 pub trait BinomiallyExtendable<const D: usize>: Field + Sized {
     const W: Self;
+
+    // DTH_ROOT = W^((n - 1)/D).
+    // n is the order of base field.
+    // Only works when exists k such that n = kD + 1.
+    const DTH_ROOT: Self;
+
     fn ext_multiplicative_group_generator() -> [Self; D];
 }
+
+pub trait Frobenius<F: BinomiallyExtendable<D>, const D: usize>:
```

Comment:
Continuing the discussion [here](https://github.com/Plonky3/Plonky3/pull/116#discussion_r1333879923),

You pointed out that generalizing the Frobenius API would make it harder to provide a good default impl. But I think it would be fine to have no default impl in the trait, and move this OEF Frobenius logic into `BinomialExtensionField`'s impl of the trait. What do you think?

Reply:
I see. I make the frobenuis trait to be defined for the `ExtensionField`(in case we have non-binomial extension in the future) and move the current implementation to `BinomialExtensionFields`, is this desired?

Reply:
Yeah I think that sounds good. In the most general form it could be `trait Frobenius: AbstractField` (with no default impls), but `trait Frobenius: ExtensionField` seems reasonable also, it seems easy to change later if something comes up where we want the extra generality.